src/components/Icons/SunIcon.js

```diff
@@ -14,39 +14,44 @@ const SunIcon = () => {
       viewBox='0 0 24 24'
       fill='none'
       xmlns='http://www.w3.org/2000/svg'
+      data-glean-label='Sun Theme'
     >
-      <circle cx='11.9998' cy='11.9998' r='5.75375' fill='currentColor' />
+      <circle cx='11.9998' cy='11.9998' r='5.75375' fill='currentColor' data-glean-label='Sun Theme'/>
```

Comment:
Have you tried, instead, setting the label [here](https://github.com/mozilla/debug-ping-view/blob/f7ae9387ddefb73c224c6f46290e54a72082e29c/src/components/Theme/ThemeToggle.js#L17)?

Reply:
Yes, I have. The click events never get fired on `<button>` element that you linked.

Reply:
It's quite a bit odd that we need to tag all the sub-elements that make a React element: this is not going to be possible when using other things we don't own. I wonder if the problem lies elsewhere.

Reply:
It is weird that events aren't fired on the `button`. I agree that if we have to tag elements really far down in the DOM tree it might get really hard for users to figure out how to tag their elements.

I also wonder if there is someway we can fix our side so that we can get the button element clicks to register

Reply:
Ae are capturing clicks on those elements that have `data-glean-*` attributes set.

One approach that @badboy suggested is to add "click" event listeners for all those elements that have `data-glean-*` attributes set (instead of a general click listener on the entire document and then filtering which ones have these attributes set) in the document and using the [event bubbling](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Building_blocks/Events#event_bubbling) to capture events on the parent of the outermost svg element.

This might work but I will have to test it to say for sure.

Reply:
For the specific issue above, I was able to get the button click working by setting `pointerEvents: 'none'` on the SVG elements instead.

It would be nice if we could somehow walk the tree of elements to check for the `data-glean-label` rather than having to rely on that one specific element having it. Event bubbling looks like what we may need, I am going to look into any ways to see if we can walk the DOM tree at all

Reply:
> For the specific issue above, I was able to get the button click working by setting `pointerEvents: 'none'` on the SVG elements instead.

I think this is the right path: let's not over-tweak this for this specific use-case, but rather just add the pointerEvents properties to the icons we have committed in tree.